### PR TITLE
#89: Implement setProperties request.

### LINKED ISSUE #89: Implement setProperties request
## Acceptance criteria

* Users can change schema name with `ALTER VIRTUAL SCHEMA ... SET`
* Users can change table filter with `ALTER VIRTUAL SCHEMA ... SET`

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #89.

### CODE REVIEW

File: doc/changes/changes_1.0.0.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Code name: Table filter
 
 Version 1.0.0 of Row Level Security (Lua) adds a table filter controlled by the `TABLE_FILTER` property. This allows users to selectively include tables from the source schema into the Virtual Schema. It helps reducing the projected tables on a need-to-know basis and speeds up meta-data scanning, since only the listed tables are scanned instead of all in the schema.
 
+You can now also use `ALTER VIRTUAL SCHEMA ... SET` to change the properties of an existing Virtual schema.
```
Comment (kaklakariada) [inline doc/changes/changes_1.0.0.md:7]:
```suggestion
You can now also use `ALTER VIRTUAL SCHEMA ... SET ...` to change the properties of an existing Virtual schema.
```

File: doc/user_guide/user_guide.md [hunk 2]
```diff
@@ -427,6 +427,21 @@ CREATE VIRTUAL SCHEMA RLS_VIRTUAL_SCHEMA
 
 Spaces around the table names are ignored.
 
+### Changing the Properties of an Existing Virtual Schema
+
+While you could in theory drop and re-create an Virtual Schema, there is a more convenient way to apply changes in the adapter properties.
+
+Use `ALTER VIRTUAL SCHEMA ... SET` to update an existing Virtual Schema to update the properties and with them an existing Virtual Schema.
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:434]:
```suggestion
Use `ALTER VIRTUAL SCHEMA ... SET ...` to update the properties of an existing Virtual Schema.
```

File: doc/user_guide/user_guide.md [hunk 3]
```diff
@@ -427,6 +427,21 @@ CREATE VIRTUAL SCHEMA RLS_VIRTUAL_SCHEMA
...
+
+Example:
+
+```sql
+ALTER VIRTUAL SCHEMA RLS_VIRTUAL_SCHEMA
+SET SCHEMA_NAME = '<new schema name>'
+```
+
+You can for example change the `SCHEMA_NAME` property to point the Virtual Schema to a new source schema.
```
Comment (kaklakariada) [inline doc/user_guide/user_guide.md:443]: ... or add a table filter?

File: src/test/java/com/exasol/MetadataReadingIT.java [hunk 4]
```diff
@@ -164,4 +164,19 @@ void testTableFilter() throws SQLException {
...
+    void testSwitchingSourceSchemaWithAlterVirtualSchema() throws SQLException {
+        final Schema schemaA = createSchema("SCHEMA_SET_PROPS_A");
+        schemaA.createTable("T", "C1", "VARCHAR(10)").insert("Hello");
+        final Schema schemaB = createSchema("SCHEMA_SET_PROPS_B");
+        schemaB.createTable("T", "C1", "DATE").insert("1970-01-01");
+        final VirtualSchema virtualSchema = createVirtualSchema(schemaA);
+        final User user = createUserWithVirtualSchemaAccess("USER_SET_PROPS", virtualSchema);
+        assertRlsQueryWithUser("SELECT * FROM " + virtualSchema.getName() + ".T", user, table().row("Hello").matches());
+        execute("ALTER VIRTUAL SCHEMA " + virtualSchema.getName() + " SET SCHEMA_NAME = '" + schemaB.getName() + "'");
+        final Date expectedDate = Date.valueOf("1970-01-01");
```
Comment (kaklakariada) [inline src/test/java/com/exasol/MetadataReadingIT.java:178]: String `Hello` and date `1970-01-01` are duplicated.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/MetadataReadingIT.java (hunk 4 above)
Comment (redcatbear) [reply src/test/java/com/exasol/MetadataReadingIT.java:178]: Fixed.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**